console/collections/src/merkle_tree/tests/batch.rs

```diff
@@ -0,0 +1,675 @@
+        type PH = Poseidon<CurrentEnvironment, 2>;
+
+        let leaf_hasher = LH::setup("AleoMerkleTreeTest0")?;
+        let path_hasher = PH::setup("AleoMerkleTreeTest1")?;
+
+        for i in 0..ITERATIONS {
+            for j in 0..ITERATIONS {
+                // Determine the leaves and additional leaves.
+                let num_leaves = core::cmp::min(2u128.pow(DEPTH as u32), i);
+                let num_additional_leaves =
+                    core::cmp::min(num_leaves, core::cmp::min(2u128.pow(DEPTH as u32) - num_leaves, j));
+
+                // Check the Merkle tree.
+                check_merkle_tree::<CurrentEnvironment, LH, PH, DEPTH>(
+                    &leaf_hasher,
+                    &path_hasher,
+                    &(0..num_leaves).map(|_| vec![Uniform::rand(rng)]).collect::<Vec<_>>(),
+                    &(0..num_additional_leaves)
+                        .rev()
+                        .map(|i| ((i % num_leaves) as usize, vec![Uniform::rand(rng)]))
+                        .collect::<Vec<_>>(),
+                )?;
+            }
+        }
+        Ok(())
+    }
+
+    let mut rng = TestRng::default();
+
+    // Ensure DEPTH = 0 fails.
+    assert!(run_test::<0>(&mut rng).is_err());
+    // Spot check important depths.
+    assert!(run_test::<1>(&mut rng).is_ok());
+    assert!(run_test::<2>(&mut rng).is_ok());
+    assert!(run_test::<3>(&mut rng).is_ok());
+    assert!(run_test::<4>(&mut rng).is_ok());
+    assert!(run_test::<5>(&mut rng).is_ok());
+    assert!(run_test::<6>(&mut rng).is_ok());
+    assert!(run_test::<7>(&mut rng).is_ok());
+    assert!(run_test::<8>(&mut rng).is_ok());
+    assert!(run_test::<9>(&mut rng).is_ok());
+    assert!(run_test::<10>(&mut rng).is_ok());
+    assert!(run_test::<15>(&mut rng).is_ok());
+    assert!(run_test::<16>(&mut rng).is_ok());
+    assert!(run_test::<17>(&mut rng).is_ok());
+    assert!(run_test::<31>(&mut rng).is_ok());
+    assert!(run_test::<32>(&mut rng).is_ok());
+    assert!(run_test::<64>(&mut rng).is_ok());
+    Ok(())
+}
+
+#[test]
+fn test_merkle_tree_bhp_batch_update_is_consistent() -> Result<()> {
+    fn run_test<const DEPTH: u8>(rng: &mut TestRng) -> Result<()> {
+        type LH = BHP1024<CurrentEnvironment>;
+        type PH = BHP512<CurrentEnvironment>;
+
+        let leaf_hasher = LH::setup("AleoMerkleTreeTest0")?;
+        let path_hasher = PH::setup("AleoMerkleTreeTest1")?;
+
+        for _ in 0..ITERATIONS {
+            // Determine the leaves and additional leaves.
+            let num_leaves = 2u128.pow(DEPTH as u32);
+
+            // Construct the random updates.
+            let mut updates = (0..num_leaves)
+                .map(|_| {
+                    let index: u128 = Uniform::rand(rng);
+                    ((index % num_leaves) as usize, Field::<CurrentEnvironment>::rand(rng).to_bits_le())
+                })
+                .collect::<Vec<(usize, Vec<bool>)>>();
+            updates.sort_by(|(a, _), (b, _)| a.cmp(b));
```

Comment:
```suggestion
            updates.sort_by_key(|(a, _)| a));
```
plus the same `cmp::Reverse` note as before